src/ray/rpc/grpc_client.h

```diff
@@ -186,15 +197,20 @@ class GrpcClient {
   }
 
  private:
+  static bool TestingRpcFailureAvoidIntraNodeFailures(std::string_view server_address,
+                                                      std::string_view local_address) {
+    return ::RayConfig::instance().testing_rpc_failure_avoid_intra_node_failures() &&
+           (server_address == "127.0.0.1" || server_address == local_address);
```

Comment:
handle `::1` and `localhost` as well, create a `is_localhost()` util in network_util.h